crafters/image/ImageTorchTransformation/__init__.py

```diff
@@ -0,0 +1,165 @@
+
+    Note: Not recommended to use inherently random transforms and
+    make sure not to use transforms that do not support tensor inputs.
+
+    Example:
+        .. code-block:: python
+            from jina.hub.crafters.image.ImageTorchTransformation import ImageTorchTransformation
+            import numpy
+
+            transforms = [
+                {"CenterCrop": dict(size=(300))},
+                {"Resize": dict(size=(244, 244))},
+                "RandomVerticalFlip",
+                {"Normalize": dict(mean=(0.485, 0.456, 0.406), std=(0.229, 0.224, 0.225))},
+            ]
+
+            crafter = ImageTorchTransformation(transforms)
+            crafted_imgs = crafter.craft(image_batch)
+
+        The torch equivalent for this would be recursively apply `Compose` on each image.
+
+        .. code-block:: python
+            import torchvision.transforms as T
+
+            transforms = T.Compose(
+                [
+                    T.ToTensor(),
+                    T.CenterCrop(300),
+                    T.Resize((244, 244)),
+                    T.RandomVerticalFlip(p=1.0),
+                    T.Normalize(mean=(0.485, 0.456, 0.406), std=(0.229, 0.224, 0.225)),
+                ]
+            )
+
+            crafted_img = transforms(image_batch[0]).cpu().numpy()
+            crafted_img = np.transpose(crafted_img, (1, 2, 0))
+
+
+    :param transforms: List of transforms that are applied sequentially
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+
+    def __init__(
+        self,
+        transforms: List[Union[str, Dict[str, Dict]]],
+        *args,
+        **kwargs,
+    ):
+        """Set constructor."""
+        super().__init__(*args, **kwargs)
+        if not isinstance(transforms, list):
+            self.logger.error("The `transforms` argument has to be a list.")
+            raise ValueError
+
+        self.transforms_specification = transforms
+
+    def post_init(self):
+        from torchvision import transforms as T
+
+        transforms_list = []
+        for transform in self.transforms_specification:
+            if isinstance(transform, str):
+                tr_name, tr_kwargs = transform, {}
+            elif isinstance(transform, dict):
+                tr_name, tr_kwargs = next(iter(transform.items()))
+            else:
+                self.logger.error(
+                    "Each element of the `transform` has to be either a dict or  a string"
+                )
+                raise ValueError
+
+            try:
+                tr_class = getattr(T, tr_name)
+            except AttributeError:
+                self.logger.error(
+                    f"The transform class `{tr_name}` does not seem to exist,"
+                    "check that you have not made a typo."
+                )
+                raise ValueError
+
+            if "p" in tr_class.__init__.__code__.co_varnames:
```

Comment:
isn't there a better way to do this? Why is this needed?

Reply:
I did not find any other way to access that variable even in stacko. 
It is required to remove unwanted randomness to the processing. 
Most transforms in torchvison are 'random' i.e they take a prob, usually defaulted to 0.5 with which it is either applied or skipped. Since our usecase is inference we change it to 1.0 so it always applied. Just like in `AlbumentationsCrafter` but it has an easy `always_apply` param!

Reply:
I would simply update with p=1 as an extra argument. 

Most of the objects will just ignore except those that use it

Reply:
so just remove the if

Reply:
The stage at which I made `p`=1 is before initializing the transform and having an invalid kwarg throws error, tried it and is also testcase. Instead what I did now is to inject into the transform variables with dictionary `update`. This worked and looks neat, I guess. If `p` is used its 1 now, or its just an unwanted value it holds.